recipes/glew/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+
+class GlewConan(ConanFile):
+    name = "glew"
+    description = "The GLEW library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://github.com/nigels-com/glew"
+    topics = ("conan", "glew", "opengl", "wrangler", "loader", "binding")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("opengl/system")
+        self.requires("glu/system")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_UTILS"] = "OFF"
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        for pdb_file in glob.glob(os.path.join(self.package_folder, "lib", "*.pdb")):
+            os.remove(pdb_file)
+
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
```

Comment:
```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "GLEW"
        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
```
cmake_find_package: https://cmake.org/cmake/help/v3.14/module/FindGLEW.html
cmake_find_package_multi: namespace is also GLEW for cmake_find_package_multi, but I think that target is GLEW::glew or GLEW::glew_s :s

Reply:
shouldn't we use the component feature to set the target name `GLEW::glew` or `GLEW::glew_s` based on shared option ? Or maybe what you are proposing is already creating the `GLEW::GLEW` target, which would be conform to https://cmake.org/cmake/help/latest/module/FindGLEW.html (sorry I don't really know the components feature of conan yet, and what they allow to do)

Reply:
Something like this:
```python
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "GLEW"
        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
        self.cpp_info.components["glewlib"].names["cmake_find_package"] = "GLEW"
        self.cpp_info.components["glewlib"].names["cmake_find_package_multi"] = "glew" if self.options.shared else "glew_s"
        if self.settings.os == "Windows" and not self.options.shared:
            self.cpp_info.components["glewlib"].defines.append("GLEW_STATIC")
```